BodyweightFitnessTests/Domain/RepositoryRoutineCompanionSpec.swift

```diff
@@ -213,6 +221,78 @@ class RepositoryRoutineCompanionSpec: QuickSpec {
                     expect(companion.workoutLengthInMinutes()).to(equal(215))
                 }
             }
+
+            context("email") {
+                let unit = "kg"
+
+                it("exercisesAsCSV should return empty string") {
+                    let companion = RepositoryRoutineCompanion(RepositoryRoutine())
+                    let csv = companion.csv(weightUnit: unit)
+                    expect(csv).to(beNil())
+                }
+
+                it("exercisesAsCSV should return correct CSV") {
+                    let companion = RepositoryRoutineCompanion(self.routineCompleted)
+                    let csv = companion.csv(weightUnit: unit)
+
+                    let expectedStr =
+"""
+Date, Start Time, End Time, Workout Length, Routine, Exercise, Set Order, Weight, Weight Units, Reps, Minutes, Seconds
+Monday, 7 August 2017,13:13,15:13,2h,Bodyweight Fitness - Recommended Routine,,1,0.000000,kg,0,0,10
+Monday, 7 August 2017,13:13,15:13,2h,Bodyweight Fitness - Recommended Routine,,1,0.000000,kg,0,0,10
+Monday, 7 August 2017,13:13,15:13,2h,Bodyweight Fitness - Recommended Routine,,1,0.000000,kg,0,0,10
+
+"""
+                    let expected = expectedStr.data(using: String.Encoding.utf8, allowLossyConversion: false)
+                    expect(csv).to(equal(expected))
+                }
+
+                it("csvName should return correct filename") {
+                    let companion = RepositoryRoutineCompanion(self.routineCompleted)
+                    let csvName = companion.csvName()
+
+                    let expected = "Workout-Bodyweight Fitness-Monday, 7 August 2017 - 13:13.csv"
+                    expect(csvName).to(equal(expected))
+                }
+
+                it("emailSubject should be correct") {
+                    let companion = RepositoryRoutineCompanion(self.routineCompleted)
+                    let subject = companion.emailSubject()
+
+                    let expected = "Bodyweight Fitness workout for Monday, 7 August 2017 - 13:13"
+
+                    expect(subject).to(equal(expected))
+                }
+
+                it("emailBody should return correct body text") {
+                    let companion = RepositoryRoutineCompanion(self.routineCompleted)
+                    let body = companion.emailBody(weightUnit: unit)
+
+                    let expected =
+                    """
+Hello,
+The following is your workout in Text/HTML format (CSV attached).
+
+Workout on Monday, 7 August 2017 - 13:13.
+Last Updated at 15:13
+Workout length: 2h
+
+Bodyweight Fitness
+Recommended Routine
+
+
+Set 1, Seconds: 10
+
+
+Set 1, Seconds: 10
+
+
```

Comment:
Are exercise titles gone?

Reply:
Or maybe they are empty in the mock repository routine.

Reply:
They are empty in the sample text.

Reply:
That should be fine then, thanks for contributing! I am trying to make the code clearer and easier to contribute - still a lot to change to make it easier :-)